Failed to find GaudiAlgLib and GaudiKernel when building k4FWCore with my own Gaudi v35

Hi all, 

When I build k4FWCore using the key4hep nightlies, there is no problem. 
But when I switch to use Gaudi v35 built based on LCG98python3, the problem happens (please see below).

* Following is the configure stage, there is no error:
```
[lint@lxslc709]$ mkdir k4FWCore-build
[lint@lxslc709]$ cd k4FWCore-build/
[lint@lxslc709]$ cmake ../k4FWCore
-- The C compiler identification is GNU 8.3.0
-- The CXX compiler identification is GNU 8.3.0
-- Check for working C compiler: /cvmfs/sft.cern.ch/lcg/releases/gcc/8.3.0-cebb0/x86_64-centos7/bin/gcc
-- Check for working C compiler: /cvmfs/sft.cern.ch/lcg/releases/gcc/8.3.0-cebb0/x86_64-centos7/bin/gcc - works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /cvmfs/sft.cern.ch/lcg/releases/gcc/8.3.0-cebb0/x86_64-centos7/bin/g++
-- Check for working CXX compiler: /cvmfs/sft.cern.ch/lcg/releases/gcc/8.3.0-cebb0/x86_64-centos7/bin/g++ - works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Failed
-- Check if compiler accepts -pthread
-- Check if compiler accepts -pthread - yes
-- Found Threads: TRUE
-- Found Python: /cvmfs/sft.cern.ch/lcg/views/LCG_98python3/x86_64-centos7-gcc8-opt/bin/python3.7 (found suitable version "3.7.6", minimum required is "3.7") found components: Interpreter Development
-- Found ROOT: /cvmfs/sft.cern.ch/lcg/releases/ROOT/v6.22.00-be0a0/x86_64-centos7-gcc8-opt/cmake (found version 6.22.00)
-- Found TBB: /cvmfs/sft.cern.ch/lcg/views/LCG_98python3/x86_64-centos7-gcc8-opt/include (found version "2020.2.11102.2") found components: tbb malloc malloc_proxy
-- Found UUID: /usr/include
--   Import target: UUID::uuid
-- Found ZLIB: /cvmfs/sft.cern.ch/lcg/views/LCG_98python3/x86_64-centos7-gcc8-opt/lib/libz.so (found suitable version "1.2.11", minimum required is "1.2.11")
-- Found Rangev3: /cvmfs/sft.cern.ch/lcg/views/LCG_98python3/x86_64-centos7-gcc8-opt/include
--   Import target: Rangev3::rangev3
-- Found cppgsl: /cvmfs/sft.cern.ch/lcg/views/LCG_98python3/x86_64-centos7-gcc8-opt/include
--   Import target: cppgsl::cppgsl
-- Found nlohmann_json: /cvmfs/sft.cern.ch/lcg/views/LCG_98python3/x86_64-centos7-gcc8-opt/lib/cmake/nlohmann_json/nlohmann_jsonConfig.cmake (found version "3.6.1")
-- Found AIDA: /cvmfs/sft.cern.ch/lcg/views/LCG_98python3/x86_64-centos7-gcc8-opt/src/cpp
--   Import target: AIDA::aida
-- Found XercesC: /cvmfs/sft.cern.ch/lcg/views/LCG_98python3/x86_64-centos7-gcc8-opt/lib/libxerces-c.so (found version "3.2.3")
-- Found HepPDT: /cvmfs/sft.cern.ch/lcg/views/LCG_98python3/x86_64-centos7-gcc8-opt/include
--   Import target: HepPDT::heppdt
-- Found CppUnit: /cvmfs/sft.cern.ch/lcg/views/LCG_98python3/x86_64-centos7-gcc8-opt/include
--   Import target: CppUnit::cppunit
-- Found unwind: /cvmfs/sft.cern.ch/lcg/views/LCG_98python3/x86_64-centos7-gcc8-opt/include
--   Import target: unwind::unwind
-- Found gperftools: /cvmfs/sft.cern.ch/lcg/views/LCG_98python3/x86_64-centos7-gcc8-opt/include (found suitable version "2.7.0", minimum required is "2.7.0")
--   found components: gperftools tcmalloc tcmalloc_debug tcmalloc_and_profiler tcmalloc_minimal profiler
--   Import executable target: gperftools::pprof
-- Found Doxygen: /cvmfs/sft.cern.ch/lcg/views/LCG_98python3/x86_64-centos7-gcc8-opt/bin/doxygen (found suitable version "1.8.15", minimum required is "1.8.15") found components: doxygen dot
-- Found jemalloc: /cvmfs/sft.cern.ch/lcg/views/LCG_98python3/x86_64-centos7-gcc8-opt/lib/libjemalloc.so
--   Import target: jemalloc::jemalloc
-- Found Gaudi: /tmp/ihep.ac.cn/cepcsw/a/prototype/releases/externals/98.0.0/Gaudi/lib/cmake/Gaudi (found version 35.0)
-- Found podio: /tmp/ihep.ac.cn/cepcsw/a/prototype/releases/externals/98.0.0/podio/lib64/cmake/podio/podioConfig.cmake
-- Found EDM4HEP: /tmp/ihep.ac.cn/cepcsw/a/prototype/releases/externals/98.0.0/EDM4hep/lib64/cmake/EDM4HEP/EDM4HEPConfig.cmake
-- Configuring done
-- Generating done
-- Build files have been written to: /tmp/ihep.ac.cn/cepcsw/a/prototype/releases/externals/98.0.0/testing/k4FWCore-build
```

* Following is the make stage with VERBOSE=1:
```
[lint@lxslc709]$ make VERBOSE=1
/cvmfs/sft.cern.ch/lcg/releases/CMake/3.17.3-75516/x86_64-centos7-gcc8-opt/bin/cmake -S/tmp/ihep.ac.cn/cepcsw/a/prototype/releases/externals/98.0.0/testing/k4FWCore -B/tmp/ihep.ac.cn/cepcsw/a/prototype/releases/externals/98.0.0/testing/k4FWCore-build --check-build-system CMakeFiles/Makefile.cmake 0
/cvmfs/sft.cern.ch/lcg/releases/CMake/3.17.3-75516/x86_64-centos7-gcc8-opt/bin/cmake -E cmake_progress_start /tmp/ihep.ac.cn/cepcsw/a/prototype/releases/externals/98.0.0/testing/k4FWCore-build/CMakeFiles /tmp/ihep.ac.cn/cepcsw/a/prototype/releases/externals/98.0.0/testing/k4FWCore-build/CMakeFiles/progress.marks
make  -f CMakeFiles/Makefile2 all
make[1]: Entering directory `/tmp/ihep.ac.cn/cepcsw/a/prototype/releases/externals/98.0.0/testing/k4FWCore-build'
make  -f k4FWCore/CMakeFiles/k4FWCore.dir/build.make k4FWCore/CMakeFiles/k4FWCore.dir/depend
make[2]: Entering directory `/tmp/ihep.ac.cn/cepcsw/a/prototype/releases/externals/98.0.0/testing/k4FWCore-build'
cd /tmp/ihep.ac.cn/cepcsw/a/prototype/releases/externals/98.0.0/testing/k4FWCore-build && /cvmfs/sft.cern.ch/lcg/releases/CMake/3.17.3-75516/x86_64-centos7-gcc8-opt/bin/cmake -E cmake_depends "Unix Makefiles" /tmp/ihep.ac.cn/cepcsw/a/prototype/releases/externals/98.0.0/testing/k4FWCore /tmp/ihep.ac.cn/cepcsw/a/prototype/releases/externals/98.0.0/testing/k4FWCore/k4FWCore /tmp/ihep.ac.cn/cepcsw/a/prototype/releases/externals/98.0.0/testing/k4FWCore-build /tmp/ihep.ac.cn/cepcsw/a/prototype/releases/externals/98.0.0/testing/k4FWCore-build/k4FWCore /tmp/ihep.ac.cn/cepcsw/a/prototype/releases/externals/98.0.0/testing/k4FWCore-build/k4FWCore/CMakeFiles/k4FWCore.dir/DependInfo.cmake --color=
Dependee "/tmp/ihep.ac.cn/cepcsw/a/prototype/releases/externals/98.0.0/testing/k4FWCore-build/k4FWCore/CMakeFiles/k4FWCore.dir/DependInfo.cmake" is newer than depender "/tmp/ihep.ac.cn/cepcsw/a/prototype/releases/externals/98.0.0/testing/k4FWCore-build/k4FWCore/CMakeFiles/k4FWCore.dir/depend.internal".
Dependee "/tmp/ihep.ac.cn/cepcsw/a/prototype/releases/externals/98.0.0/testing/k4FWCore-build/k4FWCore/CMakeFiles/CMakeDirectoryInformation.cmake" is newer than depender "/tmp/ihep.ac.cn/cepcsw/a/prototype/releases/externals/98.0.0/testing/k4FWCore-build/k4FWCore/CMakeFiles/k4FWCore.dir/depend.internal".
Scanning dependencies of target k4FWCore
make[2]: Leaving directory `/tmp/ihep.ac.cn/cepcsw/a/prototype/releases/externals/98.0.0/testing/k4FWCore-build'
make  -f k4FWCore/CMakeFiles/k4FWCore.dir/build.make k4FWCore/CMakeFiles/k4FWCore.dir/build
make[2]: Entering directory `/tmp/ihep.ac.cn/cepcsw/a/prototype/releases/externals/98.0.0/testing/k4FWCore-build'
[  7%] Building CXX object k4FWCore/CMakeFiles/k4FWCore.dir/src/PodioDataSvc.cpp.o
cd /tmp/ihep.ac.cn/cepcsw/a/prototype/releases/externals/98.0.0/testing/k4FWCore-build/k4FWCore && /cvmfs/sft.cern.ch/lcg/releases/gcc/8.3.0-cebb0/x86_64-centos7/bin/g++  -Dk4FWCore_EXPORTS -I/tmp/ihep.ac.cn/cepcsw/a/prototype/releases/externals/98.0.0/testing/k4FWCore/k4FWCore/include -isystem /tmp/ihep.ac.cn/cepcsw/a/prototype/releases/externals/98.0.0/podio/include -isystem /cvmfs/sft.cern.ch/lcg/releases/ROOT/v6.22.00-be0a0/x86_64-centos7-gcc8-opt/include  -fPIC   -std=gnu++17 -o CMakeFiles/k4FWCore.dir/src/PodioDataSvc.cpp.o -c /tmp/ihep.ac.cn/cepcsw/a/prototype/releases/externals/98.0.0/testing/k4FWCore/k4FWCore/src/PodioDataSvc.cpp
[ 14%] Building CXX object k4FWCore/CMakeFiles/k4FWCore.dir/src/KeepDropSwitch.cpp.o
cd /tmp/ihep.ac.cn/cepcsw/a/prototype/releases/externals/98.0.0/testing/k4FWCore-build/k4FWCore && /cvmfs/sft.cern.ch/lcg/releases/gcc/8.3.0-cebb0/x86_64-centos7/bin/g++  -Dk4FWCore_EXPORTS -I/tmp/ihep.ac.cn/cepcsw/a/prototype/releases/externals/98.0.0/testing/k4FWCore/k4FWCore/include -isystem /tmp/ihep.ac.cn/cepcsw/a/prototype/releases/externals/98.0.0/podio/include -isystem /cvmfs/sft.cern.ch/lcg/releases/ROOT/v6.22.00-be0a0/x86_64-centos7-gcc8-opt/include  -fPIC   -std=gnu++17 -o CMakeFiles/k4FWCore.dir/src/KeepDropSwitch.cpp.o -c /tmp/ihep.ac.cn/cepcsw/a/prototype/releases/externals/98.0.0/testing/k4FWCore/k4FWCore/src/KeepDropSwitch.cpp
[ 21%] Linking CXX shared library libk4FWCore.so
cd /tmp/ihep.ac.cn/cepcsw/a/prototype/releases/externals/98.0.0/testing/k4FWCore-build/k4FWCore && /cvmfs/sft.cern.ch/lcg/releases/CMake/3.17.3-75516/x86_64-centos7-gcc8-opt/bin/cmake -E cmake_link_script CMakeFiles/k4FWCore.dir/link.txt --verbose=1
/cvmfs/sft.cern.ch/lcg/releases/gcc/8.3.0-cebb0/x86_64-centos7/bin/g++ -fPIC   -shared -Wl,-soname,libk4FWCore.so -o libk4FWCore.so CMakeFiles/k4FWCore.dir/src/PodioDataSvc.cpp.o CMakeFiles/k4FWCore.dir/src/KeepDropSwitch.cpp.o   -L/tmp/ihep.ac.cn/cepcsw/a/prototype/releases/externals/98.0.0/testing/k4FWCore-build/.plugins  -Wl,-rpath,/tmp/ihep.ac.cn/cepcsw/a/prototype/releases/externals/98.0.0/testing/k4FWCore-build/.plugins:/tmp/ihep.ac.cn/cepcsw/a/prototype/releases/externals/98.0.0/podio/lib64:/cvmfs/sft.cern.ch/lcg/releases/ROOT/v6.22.00-be0a0/x86_64-centos7-gcc8-opt/lib: -lGaudiAlgLib -lGaudiKernel /tmp/ihep.ac.cn/cepcsw/a/prototype/releases/externals/98.0.0/podio/lib64/libpodioRootIO.so /cvmfs/sft.cern.ch/lcg/releases/ROOT/v6.22.00-be0a0/x86_64-centos7-gcc8-opt/lib/libTree.so /cvmfs/sft.cern.ch/lcg/releases/ROOT/v6.22.00-be0a0/x86_64-centos7-gcc8-opt/lib/libNet.so /cvmfs/sft.cern.ch/lcg/releases/ROOT/v6.22.00-be0a0/x86_64-centos7-gcc8-opt/lib/libRIO.so /cvmfs/sft.cern.ch/lcg/releases/ROOT/v6.22.00-be0a0/x86_64-centos7-gcc8-opt/lib/libMathCore.so /cvmfs/sft.cern.ch/lcg/releases/ROOT/v6.22.00-be0a0/x86_64-centos7-gcc8-opt/lib/libImt.so /cvmfs/sft.cern.ch/lcg/releases/ROOT/v6.22.00-be0a0/x86_64-centos7-gcc8-opt/lib/libThread.so /cvmfs/sft.cern.ch/lcg/releases/ROOT/v6.22.00-be0a0/x86_64-centos7-gcc8-opt/lib/libCore.so -lpthread /tmp/ihep.ac.cn/cepcsw/a/prototype/releases/externals/98.0.0/podio/lib64/libpodio.so
/cvmfs/sft.cern.ch/lcg/releases/binutils/2.30-e5b21/x86_64-centos7/bin/ld: cannot find -lGaudiAlgLib
/cvmfs/sft.cern.ch/lcg/releases/binutils/2.30-e5b21/x86_64-centos7/bin/ld: cannot find -lGaudiKernel
collect2: error: ld returned 1 exit status
make[2]: *** [k4FWCore/libk4FWCore.so] Error 1
make[2]: Leaving directory `/tmp/ihep.ac.cn/cepcsw/a/prototype/releases/externals/98.0.0/testing/k4FWCore-build'
make[1]: *** [k4FWCore/CMakeFiles/k4FWCore.dir/all] Error 2
make[1]: Leaving directory `/tmp/ihep.ac.cn/cepcsw/a/prototype/releases/externals/98.0.0/testing/k4FWCore-build'
make: *** [all] Error 2
[lint@lxslc709]$ which cmake
/cvmfs/sft.cern.ch/lcg/views/LCG_98python3/x86_64-centos7-gcc8-opt/bin/cmake
[lint@lxslc709]$ cmake --version
cmake version 3.17.3
```

My current solution is adding the `Gaudi::` prefix to the targets in the CMakeLists.txt files:
```diff
[lint@lxslc709]$ git diff
diff --git a/k4FWCore/CMakeLists.txt b/k4FWCore/CMakeLists.txt
index 55049ff..84bf189 100644
--- a/k4FWCore/CMakeLists.txt
+++ b/k4FWCore/CMakeLists.txt
@@ -10,7 +10,7 @@ gaudi_install(SCRIPTS)
 gaudi_add_library(k4FWCore
                  SOURCES src/PodioDataSvc.cpp
               src/KeepDropSwitch.cpp
-                  LINK GaudiAlgLib GaudiKernel podio::podioRootIO ROOT::Core ROOT::RIO ROOT::Tree
+                  LINK Gaudi::GaudiAlgLib Gaudi::GaudiKernel podio::podioRootIO ROOT::Core ROOT::RIO ROOT::Tree
                   )
 target_include_directories(k4FWCore PUBLIC
   $<BUILD_INTERFACE:${CMAKE_CURRENT_LIST_DIR}/include>
@@ -22,7 +22,7 @@ target_include_directories(k4FWCore PUBLIC
 file(GLOB k4fwcore_plugin_sources components/*.cpp)
 gaudi_add_module(k4FWCorePlugins
                  SOURCES ${k4fwcore_plugin_sources}
-                 LINK GaudiAlgLib GaudiKernel k4FWCore ROOT::Core ROOT::RIO ROOT::Tree)
+                 LINK Gaudi::GaudiAlgLib Gaudi::GaudiKernel k4FWCore ROOT::Core ROOT::RIO ROOT::Tree)
 target_include_directories(k4FWCorePlugins PUBLIC
   $<BUILD_INTERFACE:${CMAKE_CURRENT_LIST_DIR}/include>
   $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>)
diff --git a/test/k4TestFWCore/CMakeLists.txt b/test/k4TestFWCore/CMakeLists.txt
index 0101584..9f212b0 100644
--- a/test/k4TestFWCore/CMakeLists.txt
+++ b/test/k4TestFWCore/CMakeLists.txt
@@ -10,7 +10,7 @@ find_package(EDM4HEP)
 file(GLOB k4testfwcore_plugin_sources src/components/*.cpp)
 gaudi_add_module(k4TestFWCorePlugins
                  SOURCES ${k4testfwcore_plugin_sources}
-                 LINK GaudiAlgLib GaudiKernel k4FWCore ROOT::Core ROOT::RIO ROOT::Tree EDM4HEP::edm4hepDict EDM4HEP::edm4hep)
+                 LINK Gaudi::GaudiAlgLib Gaudi::GaudiKernel k4FWCore ROOT::Core ROOT::RIO ROOT::Tree EDM4HEP::edm4hepDict EDM4HEP::edm4hep)


 include(CTest)
```

Do you have any suggetions? 
Thank you!
Tao

--- a/k4FWCore/CMakeLists.txt
+++ b/k4FWCore/CMakeLists.txt
@@ -10,7 +10,7 @@ gaudi_install(SCRIPTS)
 gaudi_add_library(k4FWCore
 		  SOURCES src/PodioDataSvc.cpp
               src/KeepDropSwitch.cpp
-                  LINK GaudiAlgLib GaudiKernel podio::podioRootIO ROOT::Core ROOT::RIO ROOT::Tree
+                  LINK Gaudi::GaudiAlgLib Gaudi::GaudiKernel podio::podioRootIO ROOT::Core ROOT::RIO ROOT::Tree
                   )
 target_include_directories(k4FWCore PUBLIC
   $<BUILD_INTERFACE:${CMAKE_CURRENT_LIST_DIR}/include>
@@ -22,7 +22,7 @@ target_include_directories(k4FWCore PUBLIC
 file(GLOB k4fwcore_plugin_sources components/*.cpp)
 gaudi_add_module(k4FWCorePlugins
                  SOURCES ${k4fwcore_plugin_sources}
-                 LINK GaudiAlgLib GaudiKernel k4FWCore ROOT::Core ROOT::RIO ROOT::Tree)
+                 LINK Gaudi::GaudiAlgLib Gaudi::GaudiKernel k4FWCore ROOT::Core ROOT::RIO ROOT::Tree)
 target_include_directories(k4FWCorePlugins PUBLIC
   $<BUILD_INTERFACE:${CMAKE_CURRENT_LIST_DIR}/include>
   $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>)

--- a/test/k4TestFWCore/CMakeLists.txt
+++ b/test/k4TestFWCore/CMakeLists.txt
@@ -10,7 +10,7 @@ find_package(EDM4HEP)
 file(GLOB k4testfwcore_plugin_sources src/components/*.cpp)
 gaudi_add_module(k4TestFWCorePlugins
                  SOURCES ${k4testfwcore_plugin_sources}
-                 LINK GaudiAlgLib GaudiKernel k4FWCore ROOT::Core ROOT::RIO ROOT::Tree EDM4HEP::edm4hepDict EDM4HEP::edm4hep)
+                 LINK Gaudi::GaudiAlgLib Gaudi::GaudiKernel k4FWCore ROOT::Core ROOT::RIO ROOT::Tree EDM4HEP::edm4hepDict EDM4HEP::edm4hep)
 
 
 include(CTest)

